src/plugins/expressions/common/executor/executor.ts

```diff
@@ -197,6 +201,57 @@ export class Executor<Context extends Record<string, unknown> = Record<string, u
     return execution;
   }
 
+  private walkAst(
+    ast: ExpressionAstExpression,
+    action: (fn: ExpressionFunction, link: ExpressionAstFunction) => void
+  ) {
+    const newAST = cloneDeep(ast);
+    for (const link of newAST.chain) {
+      const { function: fnName, arguments: fnArgs } = link;
+      const fn = getByAlias(this.state.get().functions, fnName);
+
+      if (fn) {
+        // if any of arguments are expressions we should migrate those first
+        link.arguments = mapValues(fnArgs, (asts, argName) => {
+          return asts.map((arg) => {
+            if (typeof arg === 'object') {
```

Comment:
will the fact that `typeof null === 'object'` cause any problems here?

Reply:
i guess it should be fine since `null` isn't a valid value for an AST arg anyway